### ISSUE TITLE
Early handle creation

### ISSUE THREAD

**Issue by cgcgcg** [kokkos/kokkos-kernels#1029]
CuBLAS and CuSparse handles are created on the first call. The creation isn't cheap, and can show up in confusing places in timing data. I can create a `Controls` object just after calling `Kokkos::initialize()`, but it might make sense to add a utility function to KokkosKernels that initializes handles for enabled TPLs.

**Comment by lucbv** [comment]
Yeah, technically you can probably already do that using this: `kokkos-kernels/src/impl/tpls/KokkosKernels_tpl_handles_def.hpp` which is the singleton that holds the cuSPARSE handle.
There is another one for cuBLAS that can also be called early, ideally that other cuBLAS thing should migrate in the more generic one...
I am planning on issuing a PR to add rocSPARSE so while I do that I can prospect a little for the functionality you want...

**Comment by lucbv** [comment]
Long awaited answer on this question: @brian-kelley added `KokkosKernels::eager_initialize()` which will initialize our TPL singletons which creates the TPL handles. Assuming this works for your use case we can probably close this?

**Comment by brian-kelley** [comment]
Yeah I think we can safely close this, the eager initialize has been in releases for a long time and is used in ``Tpetra::initialize`` automatically.
